packages/grafana-ui/src/components/Segment/Segment.tsx

```diff
@@ -18,19 +21,33 @@ export function Segment<T>({
   className,
   allowCustomValue,
   placeholder,
+  disabled,
   ...rest
 }: React.PropsWithChildren<SegmentSyncProps<T>>) {
   const [Label, width, expanded, setExpanded] = useExpandableLabel(false);
 
   if (!expanded) {
     const label = _.isObject(value) ? value.label : value;
+    const theme = useTheme();
+    const styles = getSegmentStyles(theme);
```

Comment:
```suggestion
    const styles = useStyles(getSegmentStyles);
```
If get styles function has only theme param, we can use `useStyles` hook: https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/themes/ThemeContext.tsx#L51, which would provide the theme and memoize `getStyles` calls.

Reply:
Ah, thanks!